src/typecheck/mod.rs

```diff
@@ -527,6 +528,7 @@ fn type_check_<L: Linearizer>(
                 .map_err(|err| err.into_typecheck_err(state, rt.pos))?;
 
             terms
+                .as_ref()
```

Comment:
It's the 4th time I see `as_ref().iter()` I think. Maybe `Array` should get an `iter()` function?

Reply:
Added!